Products/CMFPlone/controlpanel/browser/mail.py

```diff
@@ -37,8 +37,9 @@ def save(self):
             self.status = self.formErrorsMessage
             return False
         # keep password field
-        if (data.get('smtp_userid') is not None and
-                data.get('smtp_pass') is None):
+        smtp_user_id = (data.get('smtp_userid') or '').strip()
+        smtp_pass = (data.get('smtp_pass') or '').strip()
+        if not (smtp_user_id or smtp_pass):
```

Comment:
To match the original intent, the condition would need to be:

```
if smtp_user_id and not smtp_pass:
    del data['smtp_pass']
```

This would likely fix the failing test.